hub/api/dataset.py

```diff
@@ -81,7 +81,7 @@ def __getitem__(self, item: Union[slice, str, int]):
     def __setitem__(self, item: Union[slice, str], value):
         if isinstance(item, str):
             if isinstance(value, np.ndarray):
-                write_array(
+                add_samples_to_tensor(
```

Comment:
`write_array` used to overwrite everything, so this would take over any existing tensor. We should add a check here so that the following fails:

```
ds.image = some_data
ds.image = some_data
```
Either with a warning, or at least that it doesn't do concatenation

Reply:
i raised an error here (alongside the same error in `tensor.py`)